chapters/io/index.rst

```diff
@@ -0,0 +1,315 @@
+How to handle input and output for Flows
+=======================================
+
+This chapter explains how to provide input data to the `Flow API`. Moreover, it describes how to handle the output via
+callback functions.
+
+Input
+-----
+The input data for the Flow operations `Flow.index`, `Flow.update` and `Flow.search` can be provided in three different ways.
+In the following, the input functionality is shown using the `Flow.index` function. `Flow.update` and `Flow.search` are used analogous.
+
+#. A single Document can be sent through the Flow as shown below.
+
+    .. highlight:: python
+    .. code-block:: python
+
+        with f:
+            f.index(Document('doc content'))
+
+#. An Iterable of Documents can be provided instead. The following example shows a Generator fed into a Flow.
+
+    .. highlight:: python
+    .. code-block:: python
+
+        with f:
+            f.index((Document(f'doc content {i}') for i in range(10)))
+
+
+#. The data can also be wrapped in a parameter less function.
+
+    .. highlight:: python
+    .. code-block:: python
+
+        def input_function():
+            for i in range(10):
+                yield Document(f'doc content {i}')
+        with f:
+            f.index(input_function)
+
+#. An AsyncIterable can be used as well.
+
+    .. highlight:: python
+    .. code-block:: python
+
+        async def async_input_fn():
+            for _ in range(10):
+                yield np.random.random([4])
+                await asyncio.sleep(0.1)
+
+        with AsyncFlow() as f:
+            async f.index(input_fn()):
+
+The `Flow.delete` function accepts Document Ids instead of Documents.
+
+    .. highlight:: python
+    .. code-block:: python
+
+        with f:
+            f.delete('DOC_ID')
+
+        with f:
+            f.delete(['DOC_ID_1', 'DOC_ID_2', 'DOC_ID_3'])
+
+        def input_function_delete():
+            for i in range(10):
+                yield f'{i}'
+        with f:
+            f.delete(input_function_delete)
+
+Special Input Functions
+-----------------------
+
+There are some functions of the Flow API which simplify the input handling:
+`Flow.index_lines`, `Flow.index_ndjson`, `Flow.index_csv`, `Flow.index_files`, `Flow.index_ndarray`
+`Flow.search_lines`, `Flow.search_ndjson`, `Flow.search_csv`, `Flow.search_files` and `Flow.search_ndarray`
+
+The following examples show the `Flow.index_*` functions. Search works analogous.
+Here, a csv file is used to index Documents. The possible ways of feeding in the csv are shown.
+The function `Flow.index_lines` can be used in combination with `line_format = 'csv'`.
+A simpler version is to use `F.index_csv` where the `line_format` parameter is not needed.
+The csv data can be provided as file handler or directly as `str array`.
+    .. highlight:: csv
```

Comment:
Might need a line break here. Not sure how rst renders this one

Reply:
check